torch/_dynamo/variables/builder.py

```diff
@@ -208,6 +210,17 @@ def _wrap(self, value):
             return self.wrap_sym(value)
         if istensor(value):
             return self.wrap_tensor(value)
+        elif value.__class__ in DYNAMO_SUPPORTED_PROCESS_GROUPS:
+            # This is required for generating the LOAD_FAST instruction for the
+            # process group which is required for passing it to __compiled_fn.
+            self.tx.output.graphargs.append(GraphArg(self.get_source(), value, False))
+            return ProcessGroupVariable(
+                proxy=self.tx.output.create_graph_input(
+                    re.sub(r"[^a-zA-Z0-9]+", "_", self.name), type(value)
+                ),
```

Comment:
Hmm, we generally limit the `output.create_graph_input` calls to wrappers, as this is not really a graph input, per se. 

One issue is that this will make a `placeholder` aka fx graph input arg, which is probably not what you want, right?

Reply:
what are my options? bake the PG into the graphmodule instead? or something else?

Reply:
That depends, is this primarily for export or eager? For eager, i'd say installing as a global might not be awful. For export, we need to think of a strategy where a placeholder might end up being what you want.

Reply:
I think baking PGs into graph could be a viable option though not as-is.

Let's assume that:
- we can "decay" a process group into a tuple of integers (e.g. membership in terms of global ranks)
- we have a new set of collective APIs that understands the integer representation of groups (let's make these functional while we are at it)

With the above, I wonder if it's possible/appropriate for dynamo to:
- translate a group to the integer representation
- translate the existing collectives (e.g. dist.all_reduce) to the new calls

cc: @aazzolini @mrshenli @wanchaol

Reply:
> we can "decay" a process group into a tuple of integers (e.g. membership in terms of global ranks)

Do we need to cover the cases where there are two PGs with the same set of ranks? For that, we might need an additional identifier than just member ranks?